### ISSUE TITLE
bioconductor-dada2 1.10 segfaults

### ISSUE THREAD

**Issue by epruesse** [bioconda/bioconda-recipes#13776]
See https://github.com/benjjneb/dada2/issues/684

**Comment by epruesse** [comment]
copying https://github.com/benjjneb/dada2/issues/684#issuecomment-475985615

---
Ok, please run this with your broken conda environment activated:

```
ldd $CONDA_PREFIX/lib/R/library/data.table/libs/datatable.so | grep -vE "$CONDA_PREFIX|(lib(c|rt|dl|m|pthread)|ld-linux-x86-64|linux-vdso).so"
```

In my case, it says 

```
libgfortran.so.3 => not found
```

It looks like the package was linked to `libgfortran.so.3` while it installs `libgfortran.so.4`, which in my case leads to no lib found, in other cases might lead to a mismatching one being loaded. Not sure if this is the cause - in theory any matching `libgfortran` should work. In practice, there might be ODR violations in the library that cause the segfault.

In either case, this is a broken env.

---

**Comment by epruesse** [comment]
At the very least, `datatable.so` seems to not be linked correctly.

**Comment by epruesse** [comment]
Looks like it's a case for `conda config --set channel_priority strict` or for removing those packages from our `repodata.json`.

**Comment by thermokarst** [comment]
@epruesse, I built an env:

```bash
conda create -n dada2_debug \
  -c conda-forge \
  -c bioconda \
  -c defaults \
  --override-channels \
  bioconductor-dada2=1.10
```

Then, ran:

```bash
ldd $CONDA_PREFIX/lib/R/library/data.table/libs/datatable.so | grep -vE "$CONDA_PREFIX|(lib(c|rt|dl|m|pthread)|ld-linux-x86-64|linux-vdso).so"
```

Which returned nothing. For fun:

```bash
ldd $CONDA_PREFIX/lib/R/library/data.table/libs/datatable.so
```

returns

```bash
ldd $CONDA_PREFIX/lib/R/library/data.table/libs/datatable.so
        linux-vdso.so.1 (0x00007ffd8f8b9000)
        libR.so => /home/matthew/.conda/envs/dada2_debug/lib/R/library/data.table/libs/../../../lib/libR.so (0x00007f96bcc76000)
        libgomp.so.1 => /home/matthew/.conda/envs/dada2_debug/lib/R/library/data.table/libs/../../../../libgomp.so.1 (0x00007f96bcc50000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f96bcc0d000)
        libc.so.6 => /lib64/libc.so.6 (0x00007f96bca48000)
        libRblas.so => /home/matthew/.conda/envs/dada2_debug/lib/R/library/data.table/libs/../../../lib/./libRblas.so (0x00007f96bae09000)
        libm.so.6 => /lib64/libm.so.6 (0x00007f96bacc3000)
        libreadline.so.7 => /home/matthew/.conda/envs/dada2_debug/lib/R/library/data.table/libs/../../../lib/../../libreadline.so.7 (0x00007f96bac72000)
        libpcre.so.1 => /home/matthew/.conda/envs/dada2_debug/lib/R/library/data.table/libs/../../../lib/../../libpcre.so.1 (0x00007f96bac2b000)
        liblzma.so.5 => /home/matthew/.conda/envs/dada2_debug/lib/R/library/data.table/libs/../../../lib/../../liblzma.so.5 (0x00007f96bac02000)
        libbz2.so.1.0 => /home/matthew/.conda/envs/dada2_debug/lib/R/library/data.table/libs/../../../lib/../../libbz2.so.1.0 (0x00007f96babee000)
        libz.so.1 => /home/matthew/.conda/envs/dada2_debug/lib/R/library/data.table/libs/../../../lib/../../libz.so.1 (0x00007f96babd2000)
        librt.so.1 => /lib64/librt.so.1 (0x00007f96babc7000)
        libdl.so.2 => /lib64/libdl.so.2 (0x00007f96babc2000)
        libiconv.so.2 => /home/matthew/.conda/envs/dada2_debug/lib/R/library/data.table/libs/../../../lib/../../libiconv.so.2 (0x00007f96baada000)
        libicuuc.so.58 => /home/matthew/.conda/envs/dada2_debug/lib/R/library/data.table/libs/../../../lib/../../libicuuc.so.58 (0x00007f96ba927000)
        libicui18n.so.58 => /home/matthew/.conda/envs/dada2_debug/lib/R/library/data.table/libs/../../../lib/../../libicui18n.so.58 (0x00007f96ba6a8000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f96bd177000)
        libgfortran.so.4 => /home/matthew/.conda/envs/dada2_debug/lib/R/library/data.table/libs/../../../lib/./../../libgfortran.so.4 (0x00007f96ba578000)
        libtinfo.so.6 => /home/matthew/.conda/envs/dada2_debug/lib/R/library/data.table/libs/../../../lib/../.././libtinfo.so.6 (0x00007f96ba53b000)
        libicudata.so.58 => /home/matthew/.conda/envs/dada2_debug/lib/R/library/data.table/libs/../../../lib/../.././libicudata.so.58 (0x00007f96b8c39000)
        libstdc++.so.6 => /home/matthew/.conda/envs/dada2_debug/lib/R/library/data.table/libs/../../../lib/../.././libstdc++.so.6 (0x00007f96b8af8000)
        libgcc_s.so.1 => /home/matthew/.conda/envs/dada2_debug/lib/R/library/data.table/libs/../../../lib/../.././libgcc_s.so.1 (0x00007f96b8ae1000)
        libquadmath.so.0 => /home/matthew/.conda/envs/dada2_debug/lib/R/library/data.table/libs/../../../lib/./../../libquadmath.so.0 (0x00007f96b8aad000)
```

**Comment by epruesse** [comment]
That's looking good. The problem I had observed was a lib (that is not by necessity system provided) pulled in from `/usr/lib`, instead of coming through the conda env.

**Comment by epruesse** [comment]
This issue was resolved by the great work of @ebolyen and @thermokarst tracking down the root issue to compiler flags not set by the version of TBB that is included (vendored) in `r-rcppparallel`. It's been fixed as of `4.4.1 build 1001` and included in `4.4.2` as well.

New installs should get the fixed versions. We might want to pin this in the affected packages (`biconductor-dada2`, `bioconductor-diffustats` and `bioconductor-opossom`).

**Comment by bernt-matthias** [comment]
Seems fixed.
